mythril/laser/ethereum/transaction/transaction_models.py

```diff
@@ -40,53 +40,74 @@ def __init__(
         self.op_code = op_code
 
 
-class MessageCallTransaction:
-    """ Transaction object models an transaction"""
+class BaseTransaction:
+    """Basic transaction class holding common data."""
 
     def __init__(
         self,
         world_state: WorldState,
-        callee_account: Account,
-        caller: ExprRef,
+        callee_account: Account = None,
+        caller: ExprRef = None,
         call_data=None,
         identifier=None,
         gas_price=None,
-        call_value=None,
+        gas_limit=None,
         origin=None,
-        call_data_type=None,
         code=None,
+        call_data_type=None,
+        call_value=None,
     ):
         assert isinstance(world_state, WorldState)
-        self.id = identifier or get_next_transaction_id()
         self.world_state = world_state
-        self.callee_account = callee_account
-        self.caller = caller
-        self.call_data = (
-            Calldata(self.id, call_data)
-            if not isinstance(call_data, Calldata)
-            else call_data
-        )
+        self.id = identifier or get_next_transaction_id()
+
         self.gas_price = (
-            BitVec("gasprice{}".format(identifier), 256)
-            if gas_price is None
-            else gas_price
-        )
-        self.call_value = (
-            BitVec("callvalue{}".format(identifier), 256)
-            if call_value is None
-            else call_value
+            gas_price
+            if gas_price is not None
+            else BitVec("gasprice{}".format(identifier), 256)
         )
+        self.gas_limit = gas_limit
+
         self.origin = (
-            BitVec("origin{}".format(identifier), 256) if origin is None else origin
+            origin if origin is not None else BitVec("origin{}".format(identifier), 256)
+        )
+        self.code = code
+
+        self.caller = caller
+        self.callee_account = callee_account
+        self.call_data = (
+            call_data
+            if isinstance(call_data, Calldata)
+            else Calldata(self.id, call_data)
         )
         self.call_data_type = (
```

Comment:
I think this is an unused variable now. Can you check that (and remove it if so)